test/jdk/java/lang/annotation/LoaderLeakTest.java

```diff
@@ -26,16 +26,62 @@
  * @bug 5040740
  * @summary annotations cause memory leak
  * @author gafter
- *
- * @run shell LoaderLeak.sh
- */
+ * @library /test/lib
+ * @build jdk.test.lib.process.*
+ * @run testng LoaderLeakTest
+*/
+
+import jdk.test.lib.Utils;
+import jdk.test.lib.process.ProcessTools;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
 
-import java.net.*;
-import java.lang.ref.*;
+import java.io.FileInputStream;
+import java.lang.ref.Reference;
+import java.lang.ref.WeakReference;
+import java.nio.file.*;
 import java.util.*;
-import java.io.*;
 
-public class Main {
+public class LoaderLeakTest {
+
+    @BeforeClass
+    public void initialize() throws Exception {
+        final Path TEST_CLASSES_PATH = Paths.get(Utils.TEST_CLASSES).toAbsolutePath();
+        final Path REPOSITORY_PATH = TEST_CLASSES_PATH.resolve("classes").toAbsolutePath();
+        Files.createDirectories(REPOSITORY_PATH);
+        List<String> classes = List.of("A.class", "B.class", "C.class");
+        for (String fileName : classes) {
+            Files.move(
```

Comment:
I don't think it's a good idea to move files created and managed by `jtreg`. I'd recommend you copying them here and, in `runJavaProcess...` constructing `ProcessBuilder` youself:

```
var args = new ArrayList<String>(command.length + 1);
args.add(JDKToolFinder.getJDKTool("java"));
Collections.addAll(args, command);
var pb = new ProcessBuilder(args).directory(Paths.get(Utils.TEST_CLASSES).toFile());
```

Reply:
They are intentionally moved out of class path so that the application class loader can not find them. If they are not moved they will be loaded by the application class loader and they need to be [loaded](https://github.com/openjdk/jdk/blob/bc56a63702b8730abc1d0aebee133a5884145fa1/test/jdk/java/lang/annotation/LoaderLeakTest.java#L96) by the tests very own SimpleClassLoader in order to be able to test for the leakage.

Reply:
I understand the intention, but it can be achieved w/o messing around w/ jtreg managed directories, for example by changing `SimpleClassLoader` to not delegate the loading of `A`, `B`, `C` classes to the application CL.